src/scripts/main.js

```diff
@@ -1,3 +1,230 @@
 'use strict';
 
-// write your code here
+const button = document.querySelector('.button');
+const message = document.querySelector('.message-start');
+const lose = document.querySelector('.message-lose');
+const win = document.querySelector('.message-win');
+const rows = document.querySelectorAll('tr');
+const allCells = document.querySelectorAll('td');
+let buttonClicked = false;
+
+button.addEventListener('click', () => {
+  buttonClicked = true;
+  message.classList.add('hidden');
+
+  if (!buttonClicked) {
+    newCell();
+    newCell();
+  } else {
+    reset();
+  }
+});
+
+function isGameOver() {
+  let gameOver = true;
+
+  for (let i = 0; i < 4; i++) {
+    for (let j = 0; j < 4; j++) {
+      if (!rows[i].children[j].textContent) {
+        gameOver = false;
+        break;
+      }
+    }
+  }
+
+  if (gameOver === true) {
+    for (let i = 0; i < 4; i++) {
+      for (let j = 0; j < 3; j++) {
+        const checkCell = rows[i].children[j].textContent;
+        const nextCell = rows[i].children[j + 1].textContent;
+
+        if (checkCell === nextCell) {
+          gameOver = false;
+          break;
+        }
+      }
+    }
+  }
+
+  if (gameOver === true) {
+    for (let i = 0; i < 3; i++) {
+      for (let j = 0; j < 4; j++) {
+        const checkCell = rows[i].children[j].textContent;
+        const nextCell = rows[i + 1].children[j].textContent;
+
+        if (checkCell === nextCell) {
+          gameOver = false;
+          break;
+        }
+      }
+    }
+  }
+
+  if (gameOver === true) {
+    lose.classList.remove('hidden');
+  }
```

Comment:
The isGameOver() function only shows a game over message but doesn't actually stop the game. Consider making this function return a boolean and use that to stop the game when true.